### PR TITLE
Add shopInfo proto and updated Repository function

### PR DESCRIPTION
**Pull request by stringtheory1234**

### CODE REVIEW

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.gpay.virtualqueue.backendservice.model.Shop;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@AllArgsConstructor
+@Data
+public class ShopInfo {
+    Shop shop;
+    Long activeTokens;
```
Comment (sdumka) [inline backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java:28]: numberOfActiveTokens

Comment (stringtheory1234) [reply backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java]: changed!

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.gpay.virtualqueue.backendservice.model.Shop;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@AllArgsConstructor
+@Data
+public class ShopInfo {
+    Shop shop;
+    Long numberOfActiveTokens;
```
Comment (priytanvani) [inline backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java:28]: Nit. We prefer to use primitives: https://effectivejava.googleplex.com/third-edition#page=294.

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java [hunk 3]
```diff
@@ -109,9 +115,12 @@ public Token getNewToken(UUID shopId) {
 	// Method returns all shops keeping in mind the feature of restoring shops later
 	// on.
 	public GetShopsByShopOwnerResponse getShopsByShopOwner(String shopOwnerId) {
-		List<Shop> shops = shopMap.entrySet().stream()
-				.filter(map -> map.getValue().getShopOwnerId().equals(shopOwnerId)).map(map -> map.getValue())
-				.collect(Collectors.toList());
+		List<ShopInfo> shops = new ArrayList<>();
+		shopMap.entrySet().stream().forEach(shop -> {
+			if (shop.getValue().getShopOwnerId().equals(shopOwnerId)) {
+				shops.add(new ShopInfo(shop.getValue(), getCustomersInQueue(shop.getValue().getShopId())));
```
Comment (priytanvani) [inline backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java:121]: Nit. try to use 'collect' of java streams here.

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java (hunk 2 above)
Comment (stringtheory1234) [reply backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/proto/ShopInfo.java]: done!

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java (hunk 3 above)
Comment (stringtheory1234) [reply backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java]: done!

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java [hunk 4]
```diff
@@ -72,9 +73,11 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 		return newShop.getShopId();
 	}
 
-	public List<Shop> getAllShops() {
-		return shopMap.entrySet().stream().filter(shop -> ShopStatus.ACTIVE.equals(shop.getValue().getStatus()))
-				.map(Map.Entry::getValue).collect(Collectors.toList());
+	public List<ShopInfo> getAllShops() {
+		List<ShopInfo> shops = new ArrayList<>();
+		shops = shopMap.values().stream().filter(shop -> ShopStatus.ACTIVE.equals(shop.getStatus()))
+				.map(shop -> new ShopInfo(shop, getCustomersInQueue(shop.getShopId()))).collect(Collectors.toList());
```
Comment (priytanvani) [inline backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java:79]: Nit. Just write
return shopMap.values().stream().filter(shop -> ShopStatus.ACTIVE.equals(shop.getStatus()))
				.map(shop -> new ShopInfo(shop, getCustomersInQueue(shop.getShopId()))).collect(Collectors.toList());

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java [hunk 5]
```diff
@@ -109,9 +112,9 @@ public Token getNewToken(UUID shopId) {
 	// Method returns all shops keeping in mind the feature of restoring shops later
 	// on.
 	public GetShopsByShopOwnerResponse getShopsByShopOwner(String shopOwnerId) {
-		List<Shop> shops = shopMap.entrySet().stream()
-				.filter(map -> map.getValue().getShopOwnerId().equals(shopOwnerId)).map(map -> map.getValue())
-				.collect(Collectors.toList());
+		List<ShopInfo> shops = new ArrayList<>();
```
Comment (priytanvani) [inline backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java:115]: Nit. Remove this?

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java (hunk 4 above)
Comment (stringtheory1234) [reply backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java:78]: Yup done!

File: backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java (hunk 5 above)
Comment (stringtheory1234) [reply backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java]: Made it inline with next statement.

**Pull request merged**